android/app/src/main/java/org/pathcheck/covidsafepaths/exposurenotifications/nearby/ExposureConfigurations.kt

```diff
@@ -0,0 +1,157 @@
+                    for (i in 0 until daysSinceOnsetToInfectiousness.length()) {
+                        val mapEntry = daysSinceOnsetToInfectiousness.getJSONArray(i)
+                        daysSinceOnsetToInfectiousnessMap[mapEntry.getInt(0)] = mapEntry.getInt(1)
+                    }
+                    prefs.setDaysSinceOnsetToInfectiousness(daysSinceOnsetToInfectiousnessMap)
+
+                    prefs.setInfectiousnessWhenDaysSinceOnsetMissing(
+                        dailySummariesObject.getInt("infectiousnessWhenDaysSinceOnsetMissing")
+                    )
+                    prefs.setReportTypeWhenMissing(dailySummariesObject.getInt("reportTypeWhenMissing"))
+
+                    prefs.setTriggerThresholdWeightedDuration(json.getInt("triggerThresholdWeightedDuration"))
+
+                    Log.d(TAG, "Configuration refresh succeeded")
+                    completer.set(null)
+                } catch (exception: JSONException) {
+                    Log.e(TAG, "Failed to parse configuration: $exception")
+                    completer.setCancelled()
+                }
+            }
+            val errorListener = Response.ErrorListener { exception ->
+                Log.e(TAG, "Failed to fetch configuration: $exception")
+                completer.setCancelled()
+            }
+
+            Log.d(TAG, "Fetching configuration file from $configurationUri")
+            val request = JsonObjectRequest(configurationUri.toString(), null, responseListener, errorListener)
+            request.setShouldCache(false)
+            RequestQueueSingleton.get().add(request)
+            request
+        }
+    }
+
+    fun getDailySummariesConfig(): DailySummariesConfig {
+        val reportTypeWeights = prefs.getReportTypeWeights(DEFAULT_REPORT_TYPE_WEIGHTS)
+        val infectiousnessWeights = prefs.getInfectiousnessWeights(DEFAULT_INFECTIOUSNESS_WEIGHT)
+
+        return DailySummariesConfig.DailySummariesConfigBuilder()
+            .setAttenuationBuckets(
+                prefs.getAttenuationDurationThresholds(DEFAULT_ATTENUATION_DURATION_THRESHOLDS),
+                prefs.getAttenuationBucketWeights(DEFAULT_ATTENUATION_BUCKET_WEIGHTS)
+            )
+            .setReportTypeWeight(CONFIRMED_TEST, reportTypeWeights[0])
+            .setReportTypeWeight(CONFIRMED_CLINICAL_DIAGNOSIS, reportTypeWeights[1])
+            .setReportTypeWeight(SELF_REPORT, reportTypeWeights[2])
+            .setReportTypeWeight(RECURSIVE, reportTypeWeights[3])
+            .setInfectiousnessWeight(STANDARD, infectiousnessWeights[0])
+            .setInfectiousnessWeight(HIGH, infectiousnessWeights[1])
+            .build()
+    }
+
+    fun getDiagnosisKeysDataMapping(): DiagnosisKeysDataMapping {
+        return DiagnosisKeysDataMapping.DiagnosisKeysDataMappingBuilder()
+            .setInfectiousnessWhenDaysSinceOnsetMissing(
+                prefs.getInfectiousnessWhenDaysSinceOnsetMissing(DEFAULT_INFECTIOUSNESS_WHEN_DAY_SINCE_ONSET_MISSING)
```

Comment:
Just checking: is this supposed to be default here?

Reply:
It reads from the `SharedPreference` where we store the configuration we got from the server, and if it doesn't find a match then it uses the default value provided as a parameter